sktime/forecasting/compose/_by_group_forecaster.py

```diff
@@ -0,0 +1,526 @@
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+            There are currently no reserved values for forecasters.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        param1 = {
+            "forecasters": {
+                "smooth": NaiveForecaster(),
+                "erratic": PolynomialTrendForecaster(),
+                "intermittent": Croston(),
+                "lumpy": NaiveForecaster(),
+            },
+            "transformer": ADICVTransformer(features=["class"]),
+            "fallback_forecaster": None,
+        }
+
+        # Attempting to utilize the fallback forecaster
+        param2 = {
+            "forecasters": {},
+            "transformer": ADICVTransformer(features=["class"]),
+            "fallback_forecaster": Croston(),
+        }
+
+        params = [param1, param2]
+        return params
+
+    @property
+    def _forecasters(self):
+        """Provides an internal list of the forecasters available.
+
+        Each list item is a tuple of the format (category, forecaster)
+        where the category for which the respective forecaster is chosen
+        and the forecaster itself as the values for each tuple.
+
+        Returns
+        -------
+        forecasters : list[tuple[str, strsktime forecasters]]
+            The list of forecasters which is returned. Also includes the
+            fallback forecaster with the category: "fallback_forecaster"
+        """
+        return list(self.forecasters.values()) + [
+            "fallback_forecaster",
+            self.fallback_forecaster,
+        ]
+
+    @_forecasters.setter
+    def _forecasters(self, new_forecasters):
+        """Provide new values for the forecasters.
+
+        Parameters
+        ----------
+        new_forecasters : list[tuple[str, strsktime forecasters]]
+            The list of new forecasters to update the object's forecasters with
+        """
+        # Accepting in possible new forecasters
+        for category, forecaster in new_forecasters:
+            # We assign this in a different way
+            if category != "fallback_forecaster":
+                self.forecasters[category] = forecaster
+
+            else:
+                self.fallback_forecaster = forecaster
+
+
+# Function implementations that will be added dynamically
+# if the conditions are met. explained further above!
+def _update_predict_single(self, y, fh, X=None, update_params=True):
```

Comment:
is this redundant? Why is this needed?

Reply:
Right, there isn't exactly much of an efficient improvement there. I'll remove this